src/bundle/Resources/public/js/scripts/fieldType/base/base-rich-text.js

```diff
@@ -398,18 +398,40 @@
             const links = container.querySelectorAll('a');
             const anchorPrefix = '#';
             const protocolPrefix = 'http://';
+            const restrictedKeywords = ['ezcontent', 'ezlocation'];
 
             links.forEach((link) => {
                 const href = link.getAttribute('href');
                 const protocolPattern = /^https?:\/\//i;
+                const protocolHref = protocolPrefix.concat(href);
 
-                if (href && href.indexOf(anchorPrefix) !== 0 && !protocolPattern.test(href)) {
-                    const protocolHref = protocolPrefix.concat(href);
+                if (!href && href.indexOf(anchorPrefix) === 0) {
+                    return;
+                }
+
+                if (protocolPattern.test(href)) {
+                    return;
+                }
+
+                if (this.containsAny(href, restrictedKeywords)) {
+                    return;
+                }
 
-                    link.setAttribute('href', protocolHref);
-                    link.setAttribute('data-cke-saved-href', protocolHref);
+                link.setAttribute('href', protocolHref);
+                link.setAttribute('data-cke-saved-href', protocolHref);
+            });
+        }
+
+        containsAny(string, substrings) {
+            let isSubstringPresent = false;
+
+            substrings.forEach((substring) => {
```

Comment:
I think `Array.some` would be better.

Reply:
I agree, was concerned about browser's support, but seems to be working fine. Fixed in 1e99acc1ee158b50860f3550547cd10337ef8424.